*** /tmp/ediff0zHAHW	2021-09-30 17:23:35.751123909 +0900
--- /home/naoki/.emacs.d/elisp/bespoke-theme/bespoke-theme.el	2021-09-30 17:20:15.231109474 +0900
***************
*** 241,250 ****
  
  (defun bespoke--set-theme-variant ()
    "Set theme colors according to LIGHT or DARK variant"
!   (setq bespoke-foreground (--l/d "#282b35" "#eceff1"))
!   (setq bespoke-background (--l/d "#fffef9" "#282b35"))
  
!   (setq bespoke-modeline   (--l/d "#e3e7ef" "#3c4353"))
    (setq bespoke-highlight  (--l/d "#dbe1eb" "#444B5c"))
    (setq bespoke-inactive   (--l/d "#cbd3e1" "#525868"))
  
--- 241,250 ----
  
  (defun bespoke--set-theme-variant ()
    "Set theme colors according to LIGHT or DARK variant"
! 	(setq bespoke-foreground (--l/d "#282b35" "#eceff8"))
! 	(setq bespoke-background (--l/d "#fffef9" "#23262"))
  
! 	(setq bespoke-modeline   (--l/d "#e3e7ef" "#3c4353"))
    (setq bespoke-highlight  (--l/d "#dbe1eb" "#444B5c"))
    (setq bespoke-inactive   (--l/d "#cbd3e1" "#525868"))
  
***************
*** 794,836 ****
  
  
  ;;;;; Mu4e
!    `(mu4e-attach-number-face                      ((,class :foreground ,bespoke-strong)))
!    `(mu4e-cited-1-face                            ((,class :foreground ,bespoke-faded)))
!    `(mu4e-cited-2-face                            ((,class :foreground ,bespoke-faded)))
!    `(mu4e-cited-3-face                            ((,class :foreground ,bespoke-faded)))
!    `(mu4e-cited-4-face                            ((,class :foreground ,bespoke-faded)))
!    `(mu4e-cited-5-face                            ((,class :foreground ,bespoke-faded)))
!    `(mu4e-cited-6-face                            ((,class :foreground ,bespoke-faded)))
!    `(mu4e-cited-7-face                            ((,class :foreground ,bespoke-faded)))
!    `(mu4e-compose-header-face                     ((,class :foreground ,bespoke-faded)))
!    `(mu4e-compose-separator-face                  ((,class :foreground ,bespoke-faded)))
!    `(mu4e-contact-face                            ((,class :foreground ,bespoke-salient)))
!    `(mu4e-context-face                            ((,class :foreground ,bespoke-faded)))
!    `(mu4e-draft-face                              ((,class :foreground ,bespoke-faded)))
!    `(mu4e-flagged-face                            ((,class :foreground ,bespoke-faded)))
!    `(mu4e-footer-face                             ((,class :foreground ,bespoke-faded)))
!    `(mu4e-forwarded-face                          ((,class :inherit    default)))
!    `(mu4e-header-face                             ((,class :inherit    default)))
!    `(mu4e-header-highlight-face                   ((,class :foreground ,bespoke-subtle)))
!    `(mu4e-header-key-face                         ((,class :foreground ,bespoke-strong)))
!    `(mu4e-header-marks-face                       ((,class :foreground ,bespoke-faded)))
!    `(mu4e-header-title-face                       ((,class :foreground ,bespoke-strong)))
!    `(mu4e-header-value-face                       ((,class :inherit    default)))
!    `(mu4e-highlight-face                          ((,class :foreground ,bespoke-popout)))
!    `(mu4e-link-face                               ((,class :foreground ,bespoke-salient)))
!    `(mu4e-modeline-face                           ((,class :foreground ,bespoke-faded)))
!    `(mu4e-moved-face                              ((,class :foreground ,bespoke-faded)))
!    `(mu4e-ok-face                                 ((,class :foreground ,bespoke-faded)))
!    `(mu4e-region-code                             ((,class :foreground ,bespoke-faded)))
!    `(mu4e-replied-face                            ((,class :foreground ,bespoke-salient)))
!    `(mu4e-special-header-value-face               ((,class :inherit    default)))
!    `(mu4e-system-face                             ((,class :foreground ,bespoke-faded)))
!    `(mu4e-title-face                              ((,class :foreground ,bespoke-strong)))
!    `(mu4e-trashed-face                            ((,class :foreground ,bespoke-faded)))
!    `(mu4e-unread-face                             ((,class :foreground ,bespoke-strong)))
!    `(mu4e-url-number-face                         ((,class :foreground ,bespoke-faded)))
!    `(mu4e-view-body-face                          ((,class :inherit    default)))
!    `(mu4e-warning-face                            ((,class :foreground ,bespoke-faded)))
  
  ;;;;; Org-agenda
     `(org-agenda-calendar-event                    ((,class :inherit default)))
--- 794,836 ----
  
  
  ;;;;; Mu4e
!    ;; `(mu4e-attach-number-face                      ((,class :foreground ,bespoke-strong)))
!    ;; `(mu4e-cited-1-face                            ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-cited-2-face                            ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-cited-3-face                            ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-cited-4-face                            ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-cited-5-face                            ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-cited-6-face                            ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-cited-7-face                            ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-compose-header-face                     ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-compose-separator-face                  ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-contact-face                            ((,class :foreground ,bespoke-salient)))
!    ;; `(mu4e-context-face                            ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-draft-face                              ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-flagged-face                            ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-footer-face                             ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-forwarded-face                          ((,class :inherit    default)))
!    ;; `(mu4e-header-face                             ((,class :inherit    default)))
!    ;; `(mu4e-header-highlight-face                   ((,class :foreground ,bespoke-subtle)))
!    ;; `(mu4e-header-key-face                         ((,class :foreground ,bespoke-strong)))
!    ;; `(mu4e-header-marks-face                       ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-header-title-face                       ((,class :foreground ,bespoke-strong)))
!    ;; `(mu4e-header-value-face                       ((,class :inherit    default)))
!    ;; `(mu4e-highlight-face                          ((,class :foreground ,bespoke-popout)))
!    ;; `(mu4e-link-face                               ((,class :foreground ,bespoke-salient)))
!    ;; `(mu4e-modeline-face                           ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-moved-face                              ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-ok-face                                 ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-region-code                             ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-replied-face                            ((,class :foreground ,bespoke-salient)))
!    ;; `(mu4e-special-header-value-face               ((,class :inherit    default)))
!    ;; `(mu4e-system-face                             ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-title-face                              ((,class :foreground ,bespoke-strong)))
!    ;; `(mu4e-trashed-face                            ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-unread-face                             ((,class :foreground ,bespoke-strong)))
!    ;; `(mu4e-url-number-face                         ((,class :foreground ,bespoke-faded)))
!    ;; `(mu4e-view-body-face                          ((,class :inherit    default)))
!    ;; `(mu4e-warning-face                            ((,class :foreground ,bespoke-faded)))
  
  ;;;;; Org-agenda
     `(org-agenda-calendar-event                    ((,class :inherit default)))
